File: CodeForces/1668/C_Make_it_Increasing.py
n = int(input())
a = [int(x) for x in input().split(' ')]
# print(a)
ans = -1
for p in range(n):
    b = [0]*n
    # print(b)
    cnt = 0
    # print('p', p)
    for i in range(p+1, n):
        n = b[i-1]//a[i] + 1
        cnt += n
        b[i] = n*a[i]
    # print(list(reversed(range(p))))
    i = p-1
    print(b)
    while i>=0:
        # print(b[i+1])
        n = b[i+1]//a[i] + 1
        cnt += n
        b[i] = n*a[i]
        i -= 1

    if ans == -1 or cnt < ans:
        ans = cnt

print(ans, end='\n')

File: CodeForces/1668/C_Make_it_Increasing.cpp
// Author: Harshit Vishwakarma (Shikhar03Stark)
#include <bits/stdc++.h>
#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);
#define nl '\n'
#define ll long long int
#define ull unsigned long long int
#define dev(x) cerr << #x << " " << x << endl
using namespace std;

void pre(){
    
    return;
}

void solve(){
    int n;
    cin >> n;
    vector<ll> arr(n);
    for(auto& e: arr){
        cin >> e;
    }
    ll ans = LLONG_MAX;
    for(int  p = 0; p<n; p++){
        vector<ll> b(n, 0);
        ll cnt = 0;
        for(int i = p+1; i<n; i++){
            ll x = b[i-1]/arr[i] + 1LL;
            cnt += x;
            b[i] = x*arr[i];
        }
        for(int i = p-1; i>=0; i--){
            ll x = b[i+1]/arr[i] + 1LL;
            cnt += x;
            b[i] = x*arr[i];
        }

        ans = min(ans, cnt);
    }
    cout << ans << nl;
    return;
}

int main(){
    FASTIO
    pre();
    int T=1;
    // cin>>T;
    while(T--){
        solve();
    }
    return 0;
}